packages/block-editor/src/components/list-view/branch.js

```diff
@@ -23,39 +23,59 @@ import { isClientIdSelected } from './utils';
  * When a block is collapsed, we do not count their children as part of that total. In the current drag
  * implementation dragged blocks and their children are not counted.
  *
- * @param {Object} block            block tree
- * @param {Object} expandedState    state that notes which branches are collapsed
- * @param {Array}  draggedClientIds a list of dragged client ids
+ * @param {Object}  block               block tree
+ * @param {Object}  expandedState       state that notes which branches are collapsed
+ * @param {Array}   draggedClientIds    a list of dragged client ids
+ * @param {boolean} isExpandedByDefault flag to determine the default fallback expanded state.
  * @return {number} block count
  */
-function countBlocks( block, expandedState, draggedClientIds ) {
+function countBlocks(
+	block,
+	expandedState,
+	draggedClientIds,
+	isExpandedByDefault
```

Comment:
Wonder if this prop should instead be called `initial` and accept `collapsed`, `expanded`, or undefined.

Reply:
Right so an enum approach. 🤔

Reply:
If this is always going to be a binary status (e.g. collapsed or expanded), a non-required boolean should be fine.

If, instead, there's the chance for this to assume more states, an "enum" (i.e. a list of allowed values) is a better idea.

The `variant` prop of a `Button` is a great example of a 
"enum"-like prop. This prop was introduced to replace many boolean props (like `isPrimary`, `isSecondary`, `isTertiary`, `isLink`...)

Reply:
This isn't the main List View component, just an internal branch component. On the main component it seems to be called `expandNested`. I'm not sure why it has a different name when it seems to mean the same thing. I think List View is still exported as experimental, so the API is open to change. It could do with some tidy up as there are lots of `__experimental` props that should just be removed and treated as the default.

I personally think the expand/collapse feature should be implemented on the `TreeGrid` component, TreeGrids are supposed to support that natively. Though even then it could still be configurable through ListView.

Reply:
Good point! I believe the `TreeGrid` already has [a prop called `isExpanded`](https://github.com/WordPress/gutenberg/blob/2396f3a095d2dcf6248d60eae262a0564d13e2d9/packages/components/src/tree-grid/README.md#isexpanded) — I believe @andrewserong worked on it recently, so he may be able to confirm if the component is already good as-is, or if we're missing any features.

Reply:
> I'm not sure why it has a different name when it seems to mean the same thing

Probably a good point. One is more a "status" of a current node, the other is a state for an entire tree. 

Let's rename as we see fit to make this clearer.

Reply:
So many names :) It feels like `isExpanded` would be the best to coalesce into.